### PR TITLE
Add examples to threshold

### PR DESCRIPTION
**Pull request by timjb**

### CODE REVIEW

File: src/OpenCV/ImgProc/MiscImgTransform.hsc [hunk 1]
```diff
@@ -1167,11 +1167,49 @@ grayscale image or for removing a noise, that is, filtering out pixels with
...
+<<doc/generated/examples/threshTruncateBirds.png threshTruncateBirds>>
+<<doc/generated/examples/threshToZeroBirds.png threshToZeroBirds>>
+<<doc/generated/examples/threshToZeroInvBirds.png threshToZeroInvBirds>>
+
 <http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/miscellaneous_transformations.html#threshold OpenCV Sphinx doc>
 -}
 threshold
     :: (depth `In` [Word8, Float])
-    => ThreshValue -- ^
+    => ThreshValue
```
Comment (basvandijk) [inline src/OpenCV/ImgProc/MiscImgTransform.hsc]: We usually put in an empty `-- ^` to ensure haddock will put the individual arguments on separate lines which is easier to read. Lets leave it in.

**Comment by basvandijk** [comment]
@timjb any idea why the "Files changed" tab lists no changes?

**Comment by basvandijk** [comment]
I guess you just need to rebase (or redo the patch) on the current master.

File: src/OpenCV/ImgProc/MiscImgTransform.hsc (hunk 1 above)
Comment (timjb) [inline src/OpenCV/ImgProc/MiscImgTransform.hsc]: Sorry, thought it was a mistake.

**Comment by timjb** [comment]
@basvandijk I've rebased the patch onto the current master branch.

Here's a preview of the example images:

threshbinarybirds
threshbinaryinvbirds
threshtozerobirds
threshtozeroinvbirds
threshtruncatebirds

Maybe we should make them a bit smaller, so they don't take up that much space in the Haddocks.

**Comment by basvandijk** [comment]
Cool stuff!

**Pull request merged**